pytest_mitmproxy.py

```diff
@@ -1 +1,101 @@
 # -*- coding: utf-8 -*-
+import re
+import os
+import six
+import time
+import errno
+import shutil
+import pytest
+from contextlib import contextmanager
+from tempfile import TemporaryFile
+if six.PY3:
+    from urllib.parse import urlparse
+    from subprocess import Popen, CalledProcessError, TimeoutExpired
+elif six.PY2:
+    from urlparse import urlparse
+    from subprocess32 import Popen, CalledProcessError
+
+# clean the dumps each run
+if os.path.isdir("traffic_dumps"):
+    shutil.rmtree("traffic_dumps")
+
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--mitmproxy-filter",
+        action="store",
+        default=None,
+        help="filter specification for mitmproxy"
+    )
+
+
+@contextmanager
+def new_proxy(name, scope, flt=None):
+    ''' Start a new mitmdump process and
+    return the address for clients to connect
+
+    The captured content will be written to
+    ./<dump_dir>/<scope>/<name>
+    '''
+    dump_dir = "traffic_dumps"
+
+    try:
+        os.makedirs(os.path.join(dump_dir, scope), mode=0o755)
+    except OSError as err:
+        if err.errno != errno.EEXIST:
+            raise
+
+    dump_file = os.path.join(dump_dir, scope, name)
+
+    args = ["mitmdump", "--port=0", "--wfile={}".format(dump_file)]
+    if flt:
+        args.append(flt)
+
+    def read_all(fp):
+        fp.seek(0)
+        return fp.read().decode("utf-8")
+
+    with TemporaryFile() as stdout:
+        ps = Popen(args, stdout=stdout)
+        # Wait for status line
+        for _ in range(50):
```

Comment:
Can I get a code snippet in which you see this happeneing? Because I've tried plenty of things and regardless of what I do `select` returns before the process write another line to stdout. (